x-pack/plugins/lens/public/editor_frame_service/editor_frame/workspace_panel/workspace_panel.tsx

```diff
@@ -594,6 +601,7 @@ export const InnerWorkspacePanel = React.memo(function InnerWorkspacePanel({
     return (
       <VisualizationWrapper
         expression={localState.expressionToRender}
+        variables={getDataVariables(fromExpression(localState.expressionToRender!))}
```

Comment:
construct variables object out of the expression ast (traverse ast and find var functions with matching cache entry)